src/lobster_impl.cpp

```diff
@@ -12,175 +12,177 @@ namespace script {
-
-nfr("set_background_color", "col", "F}:4", "",
-    "sets the background color of the current cell",
-    [](StackPtr &sp, VM &vm) {
-        auto col = PopVec<float3>(sp);
-        si->SetBackgroundColor(*(uint32_t *)quantizec(col, 0.0f).data());
-    });
-
-nfr("set_text_color", "col", "F}:4", "",
-    "sets the text color of the current cell",
-    [](StackPtr &sp, VM &vm) {
-        auto col = PopVec<float3>(sp);
-        si->SetTextColor(*(uint32_t *)quantizec(col, 0.0f).data());
-    });
-
-nfr("set_text_filtered", "filtered", "B", "", "sets the text filtered of the current cell",
-    [](StackPtr &sp, VM &vm, Value filtered) {
-        si->SetTextFiltered(filtered.True());
-        return NoVal();
-    });
-
-nfr("is_text_filtered", "", "", "B", "whether the text of the current cell is filtered",
-    [](StackPtr &sp, VM &vm) { return Value(si->IsTextFiltered()); });
-
-nfr("set_border_color", "col", "F}:4", "", "sets the border color of the current grid",
-    [](StackPtr &sp, VM &vm) {
-        auto col = PopVec<float3>(sp);
-        si->SetBorderColor(*(uint32_t *)quantizec(col, 0.0f).data());
-    });
-
-nfr("set_relative_size", "s", "I", "",
-    "sets the relative size (0 is normal, -1 is smaller etc.) of the current cell",
-    [](StackPtr &sp, VM &, Value s) {
-        si->SetRelativeSize(geom::clamp(s.intval(), -10, 10));
-        return NoVal();
-    });
-
-nfr("set_style_bits", "s", "I", "",
-    "sets one or more styles (bold = 1, italic = 2, fixed = 4, underline = 8,"
-    " strikethru = 16) on the current cell.",
-    [](StackPtr &sp, VM &, Value s) {
-        si->SetStyle(s.intval());
-        return NoVal();
-    });
-
-nfr("set_status_message", "msg", "S", "",
-    "sets the status message in TreeSheets.",
-    [](StackPtr &sp, VM &vm, Value s) {
-        si->SetStatusMessage(s.sval()->strv());
-        return NoVal();
-    });
-
-nfr("get_filename_from_user", "is_save", "I", "S",
-    "gets a filename using a file dialog. empty string if cancelled.",
-    [](StackPtr &sp, VM &vm, Value is_save) {
-        return Value(vm.NewString(si->GetFileNameFromUser(is_save.True())));
-    });
-
-nfr("get_filename", "", "", "S",
-    "gets the current documents file name.",
-    [](StackPtr &sp, VM &vm) {
-        return Value(vm.NewString(si->GetFileName()));
-    });
-
-nfr("load_document", "filename", "S", "B",
-    "loads a document, and makes it the active one. returns false if failed.",
-    [](StackPtr &sp, VM &vm, Value filename) {
-        return Value(si->LoadDocument(filename.sval()->data()));
-    });
-
-nfr("set_window_size", "width,height", "II", "", "resizes the window",
-    [](StackPtr &sp, VM &vm, Value w, Value h) {
-        si->SetWindowSize(w.intval(), h.intval());
-        return NoVal();
-    });
-
-nfr("last_edit", "", "", "I",
-    "gets the timestamp of the last edit in milliseconds since the Unix/C epoch",
-    [](StackPtr &sp, VM &vm) { return Value(si->LastEdit()); });
+    nfr("goto_root", "", "", "",
```

Comment:
To not indent these was actually intentional.. they have to be inside a function, but the indentation here is not helpful when it is such a long list of declarations.